devito/passes/clusters/blocking.py

```diff
@@ -255,15 +263,15 @@ def callback(self, clusters, prefix):
                 return clusters
             skew_dim = skew_dims.pop()
 
-            # The level of a given Dimension in the hierarchy of block Dimensions, used
-            # to skew over the outer level of loops.
-            level = lambda dim: len([i for i in dim._defines if i.is_Incr])
-
             # Since we are here, prefix is skewable and nested under a
             # SEQUENTIAL loop.
             intervals = []
             for i in c.ispace:
-                if i.dim is d and level(d) <= 1:  # Skew only at level 0 or 1
+                # Skew if not Incr
```

Comment:
This is to avoid level property to be at SpaceDimension class. This way it can only be for IncrDimension.
We could have it higher in the hierarchy if DImension Classes and actually return 0 if no blocking in the game. This was the behaviour so far. Thoughts?

Reply:
I would probably have it at the highest level return 0 and only have IncrDimension return whatever is needed

Reply:
So like add in :

```python
class SpaceDimension:

    @cached_property
    def level(self):
        return 0
```
?

Reply:
All the way to `Dimension`